### PR TITLE
feat: submit order enhancements

### LINKED ISSUE #524: Transaction receipt 2nd pass: Submit Order
## Story

**As a** user who placed an Order
**I want** to see all the key details of my order transaction in my browser wallet
**So that** I can be certain I am approving the correct transaction

## Acceptance Criteria

- [x] I can see a human readable 'receipt style' view of the key details when I am requested to approve / reject a Submit Order transaction
- [ ] The view contains the necessary data enrichment to provide
  - [ ] Market name
  - [ ] Size (with correctly formatted position decimals)
  - [ ] Price (with correctly formatted asset decimals)
    - [ ] Limit order handling
    - [ ] Market order handling

## Tasks

- [x] Detailed sketch
- [x] Visual design
- [ ] Frontend tasks:
    - [ ] Implement UI
    - [ ] Integrate with backend
    - [ ] Write unit tests
- [ ] Backend tasks:
    - [ ] TBC
    - [ ] Write unit tests
- [ ] Write end to end tests
- [ ] Code review
- [ ] Team verify acceptance criteria met
- [ ] Exploratory QA testing 
- [ ] Capture bugs and new edge cases in tickets (this one or new TBD) 
- [ ] Write automated regression tests for the A/Cs (expanding on A/Cs where possible) 
- [ ] Add story and A/Cs to spec doc 
- [ ] Review with stakeholders
- [ ] Release

## Sketch

image

image

### PR DESCRIPTION
**Pull request by sam-keen**
Closes #524 

Screenshot 2023-10-02 at 17 12 31

Screenshot 2023-10-03 at 17 47 02

### CODE REVIEW

**Review by dexturr — CHANGES_REQUESTED** [review]

File: frontend/components/receipts/utils/order-table.tsx [hunk 1]
```diff
@@ -20,24 +23,105 @@ export const OrderTable = ({
...
+    getMarketById: state.getMarketById
+  }))
+
+  const market = marketId && markets.length > 0 ? getMarketById(marketId) : undefined
+  const marketDecimals = Number(market?.decimalPlaces)
+  const positionDecimals = Number(market?.positionDecimalPlaces)
+  const formattedPrice =
+    price && marketDecimals ? formatNumber(toBigNum(price, marketDecimals), marketDecimals) : undefined
+  const formattedSize =
+    size && positionDecimals ? formatNumber(toBigNum(size, positionDecimals), positionDecimals) : undefined
```
[the remark this answers is not in the document]
Comment (dexturr) [reply frontend/components/receipts/utils/order-table.tsx:60]: How we split it is another thing, let's chat and figure it out 😄

File: frontend/components/receipts/transfer/enriched-transfer-view.spec.tsx [hunk 2]
```diff
@@ -90,4 +96,17 @@ describe('EnrichedTransferView', () => {
     expect(screen.getByTestId(priceWithSymbolLocators.symbol)).toHaveTextContent('tDAI')
     expect(screen.getByTestId(vegaKeyLocators.keyName)).toHaveTextContent('External key')
   })
+
+  it('falls back to the BasicTransferView if required data is missing', () => {
+    ;(useAssetsStore as unknown as jest.Mock).mockReturnValue({
```
[the remark this answers is not in the document]
Comment (dexturr) [reply frontend/components/receipts/transfer/enriched-transfer-view.spec.tsx:101]: Created and implemented across the codebase in #571

File: specs/1118-ORDS-submit_order.md [hunk 3]
```diff
@@ -12,3 +12,14 @@
...
+
+### Market Orders
+
+- I can see 'Market price' (<a name="1118-ORDS-013" href="#1118-ORDS-013">1118-ORDS-013</a>)
+
+### Enriched Orders
+
+- I see enriched data for the market (code) when data has been loaded appropriately (<a name="1118-ORDS-011" href="#1118-ORDS-011">)
+- I see enriched data for the order size when data has been loaded appropriately (<a name="1118-ORDS-012" href="#1118-ORDS-012">1118-ORDS-012</a>)
+- I see enriched data for the order price when data has been loaded appropriately (<a name="1118-ORDS-013" href="#1118-ORDS-013">1118-ORDS-013</a>)
```
Comment (dalebennett1992) [inline specs/1118-ORDS-submit_order.md:24]: duplicate ac code, we already have 013. Not sure how you want to organise but yeah you have 013 above

File: specs/1118-ORDS-submit_order.md [hunk 4]
```diff
@@ -12,3 +12,14 @@
 
 - I can see the raw offset price of the order (<a name="1118-ORDS-008" href="#1118-ORDS-008">1118-ORDS-008</a>)
 - I can see the reference price of the order (<a name="1118-ORDS-009" href="#1118-ORDS-009">1118-ORDS-009</a>)
+
+### Market Orders
+
+- I can see 'Market price' (<a name="1118-ORDS-013" href="#1118-ORDS-013">1118-ORDS-013</a>)
```
Comment (dalebennett1992) [inline specs/1118-ORDS-submit_order.md:18]: should this be 010?

File: frontend/components/receipts/utils/order/order-size.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+describe('OrderSizeComponent', () => {
+  it('should return basic data if markets are loading or formattedSize or symbol is not defined', () => {
+    render(
+      <OrderSizeComponent marketsLoading={true} size="100" marketId="someMarketId" formattedSize="100" symbol="BTC" />
+    )
+    expect(screen.getByTestId(sizeWithTooltipLocators.sizeWithTooltip)).toBeInTheDocument()
+  })
+
+  // 1118-ORDS-012 I see enriched data for the order size when data has been loaded appropriately
```
Comment (dalebennett1992) [inline frontend/components/receipts/utils/order/order-size.spec.tsx:14]: tiny change but can we put the ac in the first line of the test? This is what we do everywhere else

File: specs/1118-ORDS-submit_order.md [hunk 6]
```diff
@@ -12,3 +12,14 @@
...
+
+### Enriched Orders
+
+- I see enriched data for the market (code) when data has been loaded appropriately (<a name="1118-ORDS-011" href="#1118-ORDS-011">)
+- I see enriched data for the order size when data has been loaded appropriately (<a name="1118-ORDS-012" href="#1118-ORDS-012">1118-ORDS-012</a>)
```
Comment (dalebennett1992) [inline specs/1118-ORDS-submit_order.md:23]: I find the wording of all these ACs a little misleading. I think 'I see enriched data for the price' makes it sounds like we will see the price AND then further data that enriches it and gives more context, which is not true. Could we word those ACs like below?
'I see the order price in the enriched data view hen data has been loaded successfully'
'I see the order size in the enriched data view when data has been loaded successfully'

File: specs/1118-ORDS-submit_order.md (hunk 4 above)
Comment (dalebennett1992) [reply specs/1118-ORDS-submit_order.md:18]: the corresponding test with this AC will also need to be updated :)

File: specs/1118-ORDS-submit_order.md [hunk 7]
```diff
@@ -12,3 +12,14 @@
...
 - I can see the raw offset price of the order (<a name="1118-ORDS-008" href="#1118-ORDS-008">1118-ORDS-008</a>)
 - I can see the reference price of the order (<a name="1118-ORDS-009" href="#1118-ORDS-009">1118-ORDS-009</a>)
+
+### Market Orders
+
+- I can see 'Market price' (<a name="1118-ORDS-013" href="#1118-ORDS-013">1118-ORDS-013</a>)
+
+### Enriched Orders
+
+- I see enriched data for the market (code) when data has been loaded appropriately (<a name="1118-ORDS-011" href="#1118-ORDS-011">)
```
[the remark this answers is not in the document]
Comment (sam-keen) [reply specs/1118-ORDS-submit_order.md:22]: No, this refers to a market code, but I'll make that more clear.

File: specs/1118-ORDS-submit_order.md (hunk 3 above)
Comment (sam-keen) [reply specs/1118-ORDS-submit_order.md:24]: looks like my find and replace didn't catch things properly! good spot, thank you

File: frontend/components/receipts/utils/order/order-size.spec.tsx (hunk 5 above)
Comment (sam-keen) [reply frontend/components/receipts/utils/order/order-size.spec.tsx:14]: Dex asked me to put them at the top of the appropriate test. Please discuss between yourselves which is the right approach

**Review by dalebennett1992 — APPROVED** [review]
can go when sonar check green
